tests/test_publish.py

```diff
@@ -143,11 +207,26 @@ def test_publish_float_2d(self, pin_query_context, published_data_reader):
             assert published_data_point.double_value == test_data_point
 
     def test_publish_bool_2d(self, pin_query_context, published_data_reader):
-        # [DigitalPattern1(ch2,ch3,ch4), DigitalPattern2(ch0)]
-        test_data = [[True, False, True], [True]]
+        # [DigitalPattern1(ch3,ch4,ch5,ch6), DigitalPattern2(ch0,ch1)]
+        test_data = [[True, False, True, False], [True, False]]
         pin_query_context.publish(test_data)
         published_data = published_data_reader.get_and_clear_published_data()
         flattened_test_data = [data_point for row in test_data for data_point in row]
         assert len(published_data) == len(flattened_test_data)
         for published_data_point, test_data_point in zip(published_data, flattened_test_data):
             assert published_data_point.boolean_value == test_data_point
+
+    def test_publish_pattern(self, standalone_tsm_context, published_data_reader):
+        (
+            pin_query_context,
+            sessions,
+            site_lists,
+        ) = standalone_tsm_context.pins_to_nidigital_sessions_for_pattern(self._PINS)
+        # [DigitalPattern1: site0, site1, site2], [DigitalPattern2: site1, site2]
+        test_data = [{0: True, 1: False, 2: True}, {1: True, 2: True}]
```

Comment:
I like this approach to defining test data because it is easier to understand than the code on line 126. I recommend using this approach in the other unit tests as well.

```suggestion
        test_data = [{1: False, 0: True, 2: True}, {1: True, 2: True}]
```

Reply:
20c4a952b4a837227c4df69a06962da870e5d30b